src/Controller/FrontPageConfigurationController.php

```diff
@@ -20,37 +22,87 @@
   * @IsGranted("ROLE_ADMIN")
   */
 final class FrontPageConfigurationController extends AbstractController {
-    public function frontPageConfig(ForumRepository $fr, SubmissionRepository $sr, ForumConfigurationRepository $fcr, EntityManager $em, Request $request) {
-        $data = new ForumConfigurationData($fcr->findSitewide());
 
-        $form = $this->createForm(ForumConfigurationType::class, $data);
+    public function frontPageConfig(
+        ForumRepository $fr,
+        SubmissionRepository $sr,
+        ForumConfigurationRepository $fcr,
+        UserRepository $ur,
+        EntityManager $em,
+        Request $request
+    ) {
+        $announcementForumName = "announcements";
+        $data = new NewForumAnnouncementData();
+
+        $form = $this->createForm(NewForumAnnouncementType::class, $data);
         $form->handleRequest($request);
 
         $message = "";
+        $messageClass = "";
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $fc = $data->toForumConfiguration();
+            // Find the user and forum for submission.
+            $user = $ur->loadUserByUsername($this->get('security.token_storage')->getToken()->getUser()->getUsername());
```

Comment:
This should be able to be simplified to $this->getUser()->getUsername();

Reply:
Oh ok, I looked around on SO and saw this was the first example of getting the username. I'll change this.